hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancerTask.java

```diff
@@ -563,6 +566,9 @@ private boolean processMoveSelection(DatanodeDetails source,
     } catch (ContainerNotFoundException e) {
       LOG.warn("Could not get container {} from Container Manager before " +
           "starting a container move", containerID, e);
+      // add source back to queue as a different container can be selected in next run.
+      findSourceStrategy.addBackSourceDataNode(source);
+      selectionCriteria.addToExcludeDueToFailContainers(moveSelection.getContainerID());
```

Comment:
How about same container in another source? do that also needs keep excluding? I think we need reconsider this container exclusion to be at source level only.

Reply:
Thanks for the review! I have created a Map of Dn and containers to be excluded for that DN. Could you please take a look at the recent changes?

Reply:
If 2 different source have same container choosen to be moved to target to reduce usages at source,
s1: c5.--> t5
s2: c5 --> t6
Now s1 for c5 had some failure, so this will avoid choosing c5 container for s1 (as expected). But old logic where comment is given, it will also make c5 container not be choosed in another source s2.

Reply:
When a container is not found in the SCM, the problem is likely outside of balancer and will affect any other sources trying to move the same container later. So I think we should exclude the container when it's a `ContainerInfo` related error like in the above case.